src/main/scala/files_metadata/infrastructure/FilesMetaPostgresRepository.scala

```diff
@@ -589,5 +589,72 @@ class FilesMetaPostgresRepository extends FilesMetaRepository {
     }
   }
 
-  override def deleteFileMeta( ownerUuid: UUID, uuid: UUID ): Unit = ???
+  override def deleteFileMeta( uuid: UUID ): Unit = {
+    val connection: Connection = pool.getConnection()
+    connection.setAutoCommit( false )
+
+    try {
+      val getArchiveStatement = connection.prepareStatement(
+        "SELECT * FROM files WHERE uuid = ?"
+      )
+      getArchiveStatement.setObject( 1, uuid )
+      val resultSet = getArchiveStatement.executeQuery()
+      if (resultSet.next()) {
+        val archive_uuid =
+          resultSet.getObject( "archive_uuid" ).asInstanceOf[UUID]
+        val deleteSharedFilesStatement = connection.prepareStatement(
```

Comment:
Don't `shared_files` get deleted automatically because of the `ON DELETE CASCADE` in the `files` table?

Reply:
Well, they don't. I think you should add the `ON DELETE CASCADE` constraint to all the tables referencing the `files` table.